apis/config/v1alpha1/projectconfig_types.go

```diff
@@ -95,6 +95,12 @@ type OpenShiftFeatureGates struct {
 
 	// OauthProxy define options for the oauth proxy feature.
 	OauthProxy OauthProxyFeatureGates `json:"oAuthProxy,omitempty"`
+
+	// DistributorServingCertsService enables OpenShift service-ca annotations on the Distributor service
+	// to use the in-platform CA and generate a TLS cert/key pair per service for in-cluster data-in-transit encryption.
+	// to use this, you need to enable TLS on the distributor and leave caName and certName empty. If you specify caName
+	// and certName, those will be used instead of the ones generated by OpenShift.
+	DistributorServingCertsService bool `json:"distributorServingCertsService,omitempty"`
```

Comment:
I'm wondering if this should be a global setting, applying to all TempoStack and TempoMonolithic instances, or instead a setting on the `TLSSpec`:
```
ServingCert bool `json:"servingCert,omitempty"`
```

I think it would be awesome if, for every TLS configuration we support in the operator, the user can choose between disabling tls, using the servingcerts service to generate it or using a custom tls secret+configmap.

This will make it available in all places where TLSSpec is used however (but it's not supported everywhere yet), so we should throw an error in the validating webhook if it's used in a place where it's not supported yet.

Reply:
I think that would be awesome sure. I can rename it but with the clarification that for now the only supported components will be the distributor and the gateway. And then in a follow up PR I can extend this.

Reply:
Yes, sounds good! Afaics in the gateway it's also only half-supported, i.e. for the gateway it currently always uses the serving cert but doesn't support custom certs.

Reply:
Yep, those are the gaps we need to fill and that is why I would like to do in a separate PR :) if not. this PR will be big